### ISSUE TITLE
[REGRESSION][BISECTED] Missing/incorrect characters after processing PDF file with pdfsizeopt

### ISSUE THREAD

**Issue by rbrito** [pts/pdfsizeopt#50]
Hi, @pts.

I found a file here generated by a dvipdfm that, when processed with pdfsizeopt, has missing symbols and/or incorrect symbols.

The file in question is the following: [wscad06.pdf](https://github.com/pts/pdfsizeopt/files/1354611/wscad06.pdf)

Using `git bisect` (thanks for the more granular commits!), I was able to discover that the first bad commit is: 26ef82d41348204d93a5421c7750c0ee6df572b5

The log of my `git bisect` is here:

```
git bisect start
# good: [4100e2038d113b9cfea587e2326d0ad287493ca7] added handling of /Encoding longer than 256 glyphs to Type1CGenerator
git bisect good 4100e2038d113b9cfea587e2326d0ad287493ca7
# bad: [cd7b8d619d34e95f04693c1cdd62e498356a3e18] updated pdfsizeopt.single
git bisect bad cd7b8d619d34e95f04693c1cdd62e498356a3e18
# good: [c5affe6ea687626d7147f56cef753f3ca458fade] added support for multiple, configurable image optimizers (--use-image-optimizer=...); this fixes https://github.com/pts/pdfsizeopt/issues/7
git bisect good c5affe6ea687626d7147f56cef753f3ca458fade
# good: [fc149a3d2b3d66519d286eb37fec9d4ce6fccd52] migrated NormalizePdfName to _EscapePdfNamesInHexTokens
git bisect good fc149a3d2b3d66519d286eb37fec9d4ce6fccd52
# good: [bded3ac6c41d3160cf8e964ebcca8d43855dd518] fixed whitespace parsing bug in front of obj definition in ParseSequentially for --stats
git bisect good bded3ac6c41d3160cf8e964ebcca8d43855dd518
# good: [f731e2de35de1c6f460cbe02d3db7151debd9ace] made trailer detection more permissive by allowing no whitespace after the trailer keyword
git bisect good f731e2de35de1c6f460cbe02d3db7151debd9ace
# good: [4d1e659cef3068aff26636a2630fd2789954247b] added removal of /Filter, /DecodeParms and /Length from non-stream objs
git bisect good 4d1e659cef3068aff26636a2630fd2789954247b
# good: [1986a7b6086f63bd4b5bfe9826dd72494d0297ab] bugfix: made Type1CConverter able to load PFB fonts by using .loadfont instead of cvx exec; this fixes https://github.com/pts/pdfsizeopt/issues/48
git bisect good 1986a7b6086f63bd4b5bfe9826dd72494d0297ab
# bad: [26ef82d41348204d93a5421c7750c0ee6df572b5] made Type1CConverter generate /Encoding from /CharStrings, so that all glyphs would be included; this fixes https://github.com/pts/pdfsizeopt/issues/49
git bisect bad 26ef82d41348204d93a5421c7750c0ee6df572b5
# good: [5205c6ba752bd637ced86974ec5b6c3beb891469] updated pdfsizeopt.single
git bisect good 5205c6ba752bd637ced86974ec5b6c3beb891469
# first bad commit: [26ef82d41348204d93a5421c7750c0ee6df572b5] made Type1CConverter generate /Encoding from /CharStrings, so that all glyphs would be included; this fixes https://github.com/pts/pdfsizeopt/issues/49
```

(You can feed that to `git bisect replay`, if you want to perform the same steps that I did. I'm also attaching the file below).

[git-bisect-log-1507093395.log](https://github.com/pts/pdfsizeopt/files/1354618/git-bisect-log-1507093395.log)

Thank you very much,

Rogério.

**Comment by pts** [comment]
Thank you very much, this detailed bug report is very useful!

**Comment by pts** [comment]
Fixed in cbb88d495ae90aea7d2596cfd94014b4729f4d7d.

The problem was that a recent change made ConvertType1FontsToType1C clobber the `/Encoding` of the font, not preserving the original.
